In [1]:
import numpy as np

In [2]:
y_true = np.array([1, 1, 0, 1, 0, 1, 1, 1, 0, 1])
y_pred = np.array([1, 0, 1, 1, 0, 0, 1, 0, 0, 0])

In [3]:
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score

Доля правильных ответов (accuracy)

In [4]:
accuracy_score(y_true, y_pred)

0.5

Точность (precision)

In [5]:
precision_score(y_true, y_pred)

0.75

Полнота (recall)

In [6]:
recall_score(y_true, y_pred)

0.42857142857142855

F1-score

In [7]:
f1_score(y_true, y_pred)

0.5454545454545454

### Вывод

В задаче прогнозирования, вернет ли клиент кредит, наиболее важным критерием является полнота (recall). Убыток в случае невозврата кредита значительно превышает прибыль в случае успешного возврата кредита. Не дать кредит человеку, который вернул бы его – не так плохо для банка, как дать кредит человеку, который его не вернет.

Accuracy модели равен 0.5. Доля правильных ответов очень низка и равна доле правильных ответов модели, возвращающей случайные результаты.

Precision равен 0.75. В отрыве от других показателей точность можно было бы считать не такой плохой.

Recall равен 0.43. Это – очень низкий результат, особенно с учетом того, что в данной задаче полнота является самым важным критерием.

F1-score равен 0.55. Это – низкий результат.

Таким образом, эта модель является очень плохой и в текущем состоянии непригодна для применения на практике.